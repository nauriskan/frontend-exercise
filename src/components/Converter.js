import React, { useState } from 'react';
import Convert from '../lib/convert'

const Converter = () => {
  const [output, setOutput] = useState('-');

  const validator = (input) => {
    if (/^\d+$/.test(input)) {
      const num = parseInt(input)

      if (num > 999999999 || num < 0) {
        setOutput('-');
      } else {
        setOutput(Convert.numToText(input));
      }
    } else if (/^[a-zA-Z ]*$/.test(input) && input.length > 0) {
      setOutput(Convert.textToNum(input));
    } else {
      setOutput('-');
    }
  };
  
  const onChange = (e) => {
    const { value } = e.target;

    validator(value);
  };

  return (
    <div>
      <input
        data-testid="input"
        type='text'
        onChange={onChange}
        style={{ width: '600px' }} />
      <div>
        <p>
            Output: <span data-testid="output">{output}</span>
        </p>
      </div>
    </div>
  )
};

export default Converter;
