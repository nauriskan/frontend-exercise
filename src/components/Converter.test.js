import React from 'react';
import { screen, render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Converter from './Converter';

describe('Converter component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Converter />)
      
    expect(asFragment(<Converter />)).toMatchSnapshot()
  });

  it('should handle invalid character input', () => {
    render(<Converter />)

    const input = screen.getByTestId('input');
    const output = screen.getByTestId('output');
      
    userEvent.type(input, '!one');
    expect(output.textContent).toBe('-');
  });

  it('should handle empty input', () => {
    render(<Converter />)

    const input = screen.getByTestId('input');
    const output = screen.getByTestId('output');
      
    userEvent.type(input, '');
    expect(output.textContent).toBe('-');
  });

  it('should handle exceeded max number input', () => {
    render(<Converter />)

    const input = screen.getByTestId('input');
    const output = screen.getByTestId('output');
      
    userEvent.type(input, '1000000000');
    expect(output.textContent).toBe('-');
  });
});
