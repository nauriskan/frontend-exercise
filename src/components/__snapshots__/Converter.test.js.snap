// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converter component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <input
      data-testid="input"
      style="width: 600px;"
      type="text"
    />
    <div>
      <p>
        Output: 
        <span
          data-testid="output"
        >
          -
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
